examples/browser-to-server/tests/test.js

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable no-console */
+import { test, expect } from '@playwright/test'
+import { playwright } from 'test-util-ipfs-example'
+import { spawn, exec } from 'child_process'
+import { existsSync } from 'fs'
+
+// Setup
+const play = test.extend({
+  ...playwright.servers()
+})
+
+async function spinUpGoLibp2p() {
+  if (!existsSync('../../examples/go-libp2p-server/go-libp2p-server')) {
+    await new Promise((resolve, reject) => {
+      exec('go build',
+        { cwd: '../../examples/go-libp2p-server' },
+        (error, stdout, stderr) => {
+          if (error) {
+            reject(error)
+            console.error(`exec error: ${error}`)
+            return
+          }
+          resolve()
+        })
+    })
+  }
+
+  const server = spawn('./go-libp2p-server', [], { cwd: '../../examples/go-libp2p-server', killSignal: 'SIGINT' })
+  server.stderr.on('data', (data) => {
+    console.log(`stderr: ${data}`, typeof data)
+  })
+  const serverAddr = await (new Promise(resolve => {
+    server.stdout.on('data', (data) => {
+      console.log(`stdout: ${data}`, typeof data)
+      if (data.includes('p2p addr:  ')) {
+        // Parse the addr out
+        resolve((String(data)).match(/p2p addr:  ([^\s]*)/)[1])
```

Comment:
```
        const addr = String(data).match(/p2p addr: ([^\s]*)/)
        if (addr.length === 0) {
          <throw error>
        }
        resolve(addr[1])
```

Reply:
I had to change the suggestion as we get (and expect) output that doesn't include the multiaddr that isn't an error case.  Additionally, we need to check for the null case when there are no matches.

```
const addr = String(data).match(/p2p addr:  ([^\s]*)/)
if (addr !== null && addr.length > 0) {
  resolve(addr[1])
}   
```